### DISCUSSION TITLE
Can such a dramatically simplified syntax model work? What do we lose in the process?

### DISCUSSION THREAD

**Discussion by bdarcus** [bdarcus/csl-next-attic#5] [Q&A]
It could be in the end it'd be better to reintroduce dedicated elements like `cs:names` and` cs:date`. What are the pros and cons of doing that versus keeping this more general?

### Pros of the more general and simple

1. (hopefully) changes would not be needed to introduce new features
2. easier to write and maintain parsers and other code
3. easier to maintain styles, since stable syntax means less concern about breakage

### Cons/Concerns

In rough order of priority/concern, with the first three far more significant than the others, which is why I've broken out separate threads for each:

1. #8 
2. #7
3. #6 
4. less clear/explicit probably (since more abstract)
5. probably a looser schema
6. ?

**Comment by denismaier** [comment]
Also: wrt validation: how can we prevent type errors when we don't have dedicated elements. E.g. this won't make sense:

```xml
<render variable="title"
             transforms="lastname-firstname-fmt"/>
```

**Reply by bdarcus** [reply]
Yes, that's partly what I mean by looser validation.

It could be addressed probably, because RNG is really powerful, along with Schematron. But not sure it's worth it.

One reason I'm working on the input model is considering the possibility that figures in here:

```xml
<render type="contributor"  variable"author" ...>
```

Or even:

```xml
<contributor variable"author" ...>
```

**Reply by denismaier** [reply]
> One reason I'm working on the input model is considering the possibility that figures in here:
> 
> ```
> <render type="contributor"  variable"author" ...>
> ```
> 
> Or even:
> 
> ```
> <contributor variable"author" ...>
> ```

What do mean here?

**Reply by bdarcus** [reply]
Just that the render node can contain the data-type, which can be used to constrain other attributes.

But I suppose the variable value can also be used to do that, which brings us back to the "maybe not worth it" point.
